rust/src/treefile.rs

```diff
@@ -2276,18 +2276,25 @@ pub(crate) fn treefile_new_client(filename: &str, basearch: &str) -> CxxResult<B
 }
 
 fn iter_etc_treefiles() -> Result<impl Iterator<Item = Result<PathBuf>>> {
-    Ok(read_dir(CLIENT_TREEFILES_DIR)?.filter_map(|res| match res {
-        Ok(e) => {
-            let path = e.path();
-            if let Some(ext) = path.extension() {
-                if ext == "yaml" {
-                    return Some(anyhow::Result::Ok(path));
+    // TODO use cap-std-ext's https://docs.rs/cap-std-ext/latest/cap_std_ext/dirext/trait.CapStdExtDirExt.html#tymethod.open_dir_optional
+    let p = Path::new(CLIENT_TREEFILES_DIR);
+    if !p.exists() {
+        return Ok(either::Left(std::iter::empty()));
```

Comment:
Hmm, I don't follow what `either` is used for here. Is this just a sneaky way to avoid having to write out the actual chained iterator type?

Reply:
By a return value of `impl T` can have *exactly one* type determined at compile time.  Now, one can return e.g. `Box<&dyn T>` which does dynamic dispatch.

More at https://internals.rust-lang.org/t/extending-impl-trait-to-allow-multiple-return-types/7921

`either` is just a convenient enum for returning at most *two* different types from a function.